src/main/java/com/garbagemule/MobArena/listeners/MAGlobalListener.java

```diff
@@ -290,22 +290,22 @@ public void playerRespawn(PlayerRespawnEvent event) {
                 return;
             }
         }
-        
+
         plugin.restoreInventory(event.getPlayer());
     }
-    
+
     public enum TeleportResponse {
         ALLOW, REJECT, IDGAF
     }
 
-    @EventHandler(priority = EventPriority.NORMAL)
+    @EventHandler(priority = EventPriority.HIGH)
```

Comment:
Changing this most certainly did fix it for your special use case. After discussing this, garbagemule and I decided to not use this. Changing the order of such an event can break more things than it would fix. Especially a volatile event like the teleport event, it can lead to messed up inventories and worse things.

I will try to give advice in the pull request